rust/arrow/src/compute/kernels/comparison.rs

```diff
@@ -41,6 +74,79 @@ where
     T: ArrowNumericType,
     F: Fn(T::Native, T::Native) -> bool,
 {
+    compare_op!(left, right, op)
+}
+
+struct RegexCache<'a> {
+    map: HashMap<&'a str, Regex>,
+}
+
+impl<'a> RegexCache<'a> {
+    fn new() -> Self {
+        Self {
+            map: HashMap::new(),
+        }
+    }
+
+    fn match_str(&'a mut self, a: &str, pat: &'a str) -> bool {
+        let re = if let Some(ref regex) = self.map.get(pat) {
+            regex
+        } else {
+            let re_pattern = pat.replace("%", ".*").replace("_", ".");
+            let re = Regex::new(&re_pattern).unwrap();
```

Comment:
Can we avoid the `unwrap` here? Or at least use `expect` with an appropriate error message?